### DISCUSSION TITLE
Does Prisma support hybrid approach

### DISCUSSION THREAD

**Discussion by mabujaber** [prisma/prisma#18965] [Q&A] [answered]
I found this modeling page in the documentation, and at the end of the documentation explains the Prisma data modeling approach
https://github.com/prisma/docs/blob/main/content/200-concepts/050-overview/100-what-is-prisma/100-data-modeling.mdx

I wonder if Prisma supports a hybrid approach, especially for legacy applications with the database created. 

Usually, in that case, we generate the model from the database, and from there, we can change the model if there are any changes needed. In this way, we followed two approaches client only and then client and migrate.

**Comment by nurul3101** [comment, marked as answer]
Hello @mabujaber 👋 

You can use [Introspection](https://www.prisma.io/docs/concepts/components/introspection) to generate prisma schema and models from your existing database. Is that what you need? Once you have the prisma schema generated your can use Prisma Client and Prisma Migrate.

**Reply by mabujaber** [reply]
That's what I'm looking for. It's worth to mention this as a third approach in documentation

**Reply by janpio** [reply]
(Note that after using INtrospection you need to baseline your migration history so that it is properly set up)

**Reply by mabujaber** [reply]
how can we do that?

**Reply by nurul3101** [reply]
Can you have a look at this section: https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/add-prisma-migrate-to-a-project#create-a-baseline-migration?
